pkg/kudoctl/cmd/init/crds.go

```diff
@@ -35,25 +37,52 @@ func installCrds(client apiextensionsclient.Interface) error {
 func installOperator(client v1beta1.CustomResourceDefinitionsGetter) error {
 	o := generateOperator()
 	_, err := client.CustomResourceDefinitions().Create(o)
-	return err
+	if !isAlreadyExistsError(err) {
+		return err
+	}
+
+	clog.V(4).Printf("crd %v already exists", o.Name)
+	return nil
+}
+
+func isAlreadyExistsError(err error) bool {
```

Comment:
```
var statusError *kerrors.StatusError
	if errors.As(err, &statusError) {
		return statusError.ErrStatus.Reason == "AlreadyExists"
	}
	return false
```

:) I spent like 45 minutes 2 days ago debugging something around error handling like this so I have this pretty fresh in my head but it's super-confusing

Basically the type you pass to As has to be `pointer to type implementing error`. For type to implement Error, it has to have the `Error()` function. So if you take a look at the way StatusError is implemented, you see 

```
func (e *StatusError) Error() string {
	return e.ErrStatus.Message
}
```

so `*StatusError` is type implementing error. So `errors.As` is expecting pointer to that type, which is `**kerrors.StatusError` :)

Reply:
perfect!  makes sense.... thanks @alenkacz